Bottom: 0bc06db4843d0b31ea555bcd062dfe1f3ded965e
Top:    4b019a99ec150531d107c14ca59d8f3976157c5d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-07 10:20:36 +0000

Workaround crash where contour is passed as NULL to poly_PreContour












---

diff --git a/src/polygon.c b/src/polygon.c
index 03e8bdd..2f82405 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -246,6 +246,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
